lib/ansible/modules/windows/win_domain_user.ps1

```diff
@@ -4,6 +4,109 @@
 
 #Requires -Module Ansible.ModuleUtils.Legacy
 
+Function Test-Credential {
+    param(
+        [String]$Username,
+        [String]$Password,
+        [String]$Domain = $null,
+        $LOGON32_PROVIDER = 0, # Default
+        $LOGON32_LOGON_TYPE = 3 # NetworkLogon
+    )
+    if (($Username.ToCharArray()) -contains [char]'@') {
+        # UserPrincipalName
+        $Username = ($Username -split '@')[0]
+        $Domain = ($Username -split '@')[1]
+    } elseif (($Username.ToCharArray()) -contains [char]'\') {
+        # Pre Win2k Account Name
+        $Username = ($Username -split '\')[0]
+        $Domain = ($Username -split '\')[1]
+    } else {
+        # No domain provided, so maybe local user?
+    }
+    $null = $Domain # Make CI-Check happy...
+
+    # More information about LogonUser at https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-logonusera
+    $platform_util = @'
+using System;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+using System.Security;
+namespace Ansible
+{
+    public class WinUserPInvoke
+    {
+        const int LOGON32_LOGON_NETWORK           = 3;
+        const int LOGON32_PROVIDER_DEFAULT        = 0; // Negotiate if domain == null and username not upn else NTLM
+        const int LOGON32_PROVIDER_WINNT40        = 2; // Negotiate
+        const int LOGON32_PROVIDER_WINNT50        = 3; // NTLM
+        // Refere to: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-erref/18d8fbe8-a967-4f1c-ae50-99ca8e491d2d
+        static int[] success_codes = new int[] {
+            0x0000052F,  // ERROR_ACCOUNT_RESTRICTION
+            0x00000530,  // ERROR_INVALID_LOGON_HOURS
+            0x00000531,  // ERROR_INVALID_WORKSTATION
+            0x00000569   // ERROR_LOGON_TYPE_GRANTED
+        };
+        static int[] failed_codes = new int[] {
+            0x0000052E,  // ERROR_LOGON_FAILURE - the user or pass was incorrect
+            0x00000006,  // ERROR_INVALID_HANDLE - the handle is invalid
```

Comment:
Why this one, sounds like a problem if you are not passing in a null ptr to LogonUser which shouldn't happen as we set that?

Reply:
Right, that should no longer be there. It was left in from debugging.